sdks/v4-sdk/src/entities/trade.ts

```diff
@@ -233,7 +233,8 @@ export class Trade<TInput extends Currency, TOutput extends Currency, TTradeType
     let outputAmount: CurrencyAmount<TOutput>
     if (tradeType === TradeType.EXACT_INPUT) {
       invariant(amount.currency.equals(route.input), 'INPUT')
-      amounts[0] = amount
+      // Account for trades that wrap as a first step
+      amounts[0] = route.pools[0].involvesCurrency(amount.currency) ? amount : amount.wrapped
```

Comment:
hiya, as we discussed in slack, consider doing this for both wrap/unwrap cases. Sorry I missed this here

https://github.com/Uniswap/sdks/blob/main/sdks/router-sdk/src/entities/mixedRoute/trade.ts#L206-L218